demo/primefaces-users-portlet/src/main/java/com/liferay/faces/demos/bean/UsersModelBean.java

```diff
@@ -88,9 +91,8 @@ public UserLazyDataModel getDataModel() {
 			int rowsPerPage = PortletHelperUtil.getPortletPreferenceAsInt(facesContext, "rowsPerPage",
 					SearchContainer.DEFAULT_DELTA);
 
-			if (!userLocalServiceTracker.isEmpty()) {
+			if (userLocalService != null) {
```

Comment:
@juangon Assuming that the ServiceTrackerCustomizer.addingService/removingService methods are called by OSGi in a thread that is different than the current request, then is there a race condition here between the OSGi thread and the request thread? If so, then perhaps the double-check idiom would be appropriate?

Reply:
What kind of check you mean?

Reply:
@juangon We typically use the double-check idiom, as shown in GenericFacesPortlet https://github.com/liferay/liferay-faces-bridge-api/blob/master/bridge-api/src/main/java/javax/portlet/faces/GenericFacesPortlet.java#L811-L831

But I think that the org.osgi.util.tracker.ServiceTracker class already has some thread-safe type code inside it.

Reply:
Yes, I think there isn't any need for more thread-safe code using ServiceTrackerCustomizer.